### PR TITLE
Create linear_search.go

### LINKED ISSUE #2205: Write a GO program for linear search
### Description

Write a GO program for linear search

Linear search is a very simple search algorithm. In this type of search, a sequential search is made over all items one by one. Every item is checked and if a match is found then that particular item is returned, otherwise the search continues till the end of the data collection.

```
Input  : [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
Target : 5

Output : 4
```
<details>
<summary>How to contribute</summary>

- Save the solution in `linear_search.go` file
- Add `linear_search.go` file in `dsa/linear-search` folder

</details>

### PR DESCRIPTION
**Pull request by MadhuS-1605**
Closes codinasion/codinasion#1624 

### What's being changed:
Added GoLang program that searches the given value in linear search and returns the position of it

### CODE REVIEW

File: dsa/linear-search/linear_search.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
+package main
+
+import "fmt"
+
+func linearSearch(A []int, n int, x int) {
+	for i := 0; i < n; i++ {
+		if A[i] == x {
+			fmt.Println("Number found at position : ", i+1)
```
Comment (victoriacheng15) [inline dsa/linear-search/linear_search.go:8]: If the array start from index 0, then it should return 4 if you are searching for 5 in [1,2,3,4,5,6,7,8,9,10]

**Review by victoriacheng15 — CHANGES_REQUESTED** [review]

File: dsa/linear-search/linear_search.go (hunk 1 above)
Comment (MadhuS-1605) [reply dsa/linear-search/linear_search.go:8]: yeah! but for the user point of view 5 is in 5th place so making it as i+1

Comment (victoriacheng15) [reply dsa/linear-search/linear_search.go:8]: Please return the index that is match the output from the issue

Comment (MadhuS-1605) [reply dsa/linear-search/linear_search.go]: done

**Review by victoriacheng15 — APPROVED** [review]

**Pull request merged**